### DISCUSSION TITLE
Install BirdNET-Pi to Proxmox LXC container

### DISCUSSION THREAD

**Discussion by dwd1** [Nachtzuster/BirdNET-Pi#111] [Show and tell]
**Install Debian 12 unprivileged in Proxmox web UI shell**
(root password = empty, minimum storage = 8GB for system and additional for bird songs)
`bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/debian.sh)"`

**After Debian 12 installed SSH to Proxmox host server and enter to the LXC shell**
(XXX = number of the container)
`pct enter XXX`

**Install BirdNET-Pi with the following commands in the LXC shell**
```
apt-get -y install sudo rsyslog
useradd -d /home/birdnet -m -s /bin/bash birdnet
usermod -aG sudo birdnet
echo "birdnet ALL=NOPASSWD: ALL" >> /etc/sudoers
su birdnet
cd
wget https://raw.githubusercontent.com/Nachtzuster/BirdNET-Pi/main/newinstaller.sh
sed -e '/sudo reboot/s/^/#/g' -i newinstaller.sh
bash newinstaller.sh
source BirdNET-Pi/birdnet/bin/activate
pip3 uninstall tflite_runtime
pip3 install tensorflow
deactivate
sudo reboot
```

**Comment by dwd1** [comment]
(reserved)

**Comment by huytman** [comment]
Perfection! Thanks so much. Now will look into migrating my DB from previous Birdnet install to this one. Thanks again.

**Reply by huytman** [reply]
was working flawlessly for 2 weeks then I started getting this error - 'BirdNET Analysis (stalled - backlog of 251 files in ~/BirdSongs/StreamData/)' This was the same error I was getting trying to install this fork of birdnet following @lloydbayley instructions from this thread - https://github.com/Nachtzuster/BirdNET-Pi/discussions/55

I had a backup of the CT (using this thread's instructions) and it's working fine. BUT. I did the update within the 'tools' menu and I'm getting the backlog error again. When I get this error - the system does not work - no birds get ID'd.

The problem with rolling back to a back-up is losing all the data gained to that point.

**Reply by dwd1** [reply]
I think if you update, the precompiled tflite_runtime is reinstalling over the manually installed tensorflow.
Try this:
```
su birdnet
cd
source BirdNET-Pi/birdnet/bin/activate
pip3 uninstall tflite_runtime
pip3 install tensorflow
deactivate
sudo reboot
```

**Reply by huytman** [reply]
that was the fix! Thanks @dwd1

**Reply by huytman** [reply]
The above code did indeed work for me - but - every few days my birdnet stops identifying and I need to re-enter the above code. No updates to the underlying software has happened - it just stops the indintifying part.

**Comment by Nachtzuster** [comment]
@dwd1 why is tflite_runtime being replaced with tensorflow?

**Reply by dwd1** [reply]
I made the description only for this reason, because it took me a long time to find the error.
Without it, the birdnet analysis service would not be able to function.

**Comment by gordy8211** [comment]
Any issues with using a mounted drive for BirdSongs? I’m unable to play any recordings from the homepage. I can play the recordings from my file server itself. The file names are not correct, they’re just random letters and numbers.
